rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -113,10 +116,36 @@ public BookKeeper(Configuration conf, MetricRegistry metrics) throws FileNotFoun
     this.ticker = ticker;
     initializeMetrics();
     initializeCache(conf, ticker);
+    cleanupOldCacheFiles(conf);
     fetchProcessor = new RemoteFetchProcessor(conf);
     fetchProcessor.startAsync();
   }
 
+  // Cleanup the cached files that were downloaded as a part of previous bookkeeper session.
+  // This makes sure we always start with a clean empty cash.
+  // TODO: We need to come up with a way to persist the files being downloaded before
+  // So that we can use that info to load those files in guava cache.
+  private void cleanupOldCacheFiles(Configuration conf)
+  {
+    if (CacheConfig.isCleanupFilesDuringStartEnabled(conf)) {
+      try {
+        int numDisks = CacheConfig.getCacheMaxDisks(conf);
+        String dirSuffix = CacheConfig.getCacheDataDirSuffix(conf);
+        List<String> dirPrefixList = CacheUtil.getDirPrefixList(conf);
+
+        for (String dirPrefix : dirPrefixList) {
+          for (int i = 0; i < numDisks; i++) {
+            java.nio.file.Path path = Paths.get(dirPrefix + i, dirSuffix, "*");
+            Files.walkFileTree(path, new DeleteFileVisitor());
```

Comment:
Wouldnt it be faster to just to `rm -rf` on all parent directories under fcache? This could get very slow in some cases.

Reply:
Done